mysqlnd_azure.c

```diff
@@ -366,13 +406,19 @@ MYSQLND_METHOD(mysqlnd_azure_data, connect)(MYSQLND_CONN_DATA ** pconn,
             enum_func_status ret = FAIL;
             MYSQLND* redirect_conneHandle = mysqlnd_init(MYSQLND_CLIENT_KNOWS_RSET_COPY_DATA, conn->persistent); //init MYSQLND but only need only MYSQLND_CONN_DATA here
             if(!redirect_conneHandle) {
+              if (MYSQLND_AZURE_G(logLevel) > 0) {
+                AZURE_LOG("DEBUG", "Cannot connet with refirection information, redirect connection handshake failed");
+              }
                 DBG_ENTER("[redirect]: init redirect_conneHandle failed");
                 if(MYSQLND_AZURE_G(enableRedirect) == REDIRECT_ON) {
                     //REDIRECT_ON, abort the original connection and return error
                     conn->m->send_close(conn);
                     SET_CLIENT_ERROR(conn->error_info, MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO, UNKNOWN_SQLSTATE, "Connection aborted because init redirection failed.");
```

Comment:
I noticed AZURE_LOG will write dupliate log as SET_CLIENT_ERROR step for this error no : MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO. So maybe we can use the identical error message in these cases, with extra message for option choice, and here missing one ERROR log "mysqnd_azure.enableRedirect=on. Connection aborted because init redirection failed."

Reply:
No need to be consistent with `MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO`, unless this number is used by some other system.
Our goal is to know the context "redirectHandle init failed && mysqlnd_azure.enableRedirection value" if some  unexpected behavior happen.
I want to know why must use this ERROR Number here?

Reply:
> I noticed AZURE_LOG will write dupliate log as SET_CLIENT_ERROR step for this error no : MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO. So maybe we can use the identical error message in these cases, with extra message for option choice, and here missing one ERROR log "mysqnd_azure.enableRedirect=on. Connection aborted because init redirection failed."

Seems there is no need to be consistent with the Error Number `MYSQLND_AZURE_ENFORCE_REDIRECT_ERROR_NO`, unless it is used by other systems.

Our goal is to get the context (mysqlnd_azure.enableRedirect value, and redirectHanle init fail or success) when unexpected scenarios happen.

If this Error Number cannot offer more information to explain the context, why we should be consistent with `SET_CLIENT_ERROR`?

Reply:
You are right. I bring this out as I think customer might check both logs, and the inconsistency might cause confusion for the same part. It's all right to me if you decide to skip some or use different message.